localstack/services/sqs/provider.py

```diff
@@ -731,10 +749,8 @@ def validate_queue_attributes(self, attributes):
                 "Invalid value for the parameter FifoQueue. Reason: Modifying queue type is not supported."
             )
 
-    # TODO: If we ever actually need to do something with this number, it needs to be part of
-    #   SQSMessage. This means changing all *put*() signatures to return the saved message.
-    def generate_sequence_number(self):
-        return _create_mock_sequence_number()
+    def next_sequence_number(self):
+        return next(global_message_sequence())
```

Comment:
is this thread-safe? I understand that the _aquisition_ of the generator object is made thread-safe by the annotation, but is the call of next() thread-safe as well? GIL is still a weak topic on my part :/

Reply:
that's actually an excellent point. they are not thread safe, and while it's not very likely there will be a race condition here, we should still make it atomic. will change!